### DISCUSSION TITLE
Automatic account linking best practices

### DISCUSSION THREAD

**Discussion by Yiling-J** [ory/kratos#3673] [Q&A] [answered]
Hi:

Need some guidance on account linking. Let's say my app suports Email/Google/Apple login, and I already enable email verification. Is it safe to link these same email accounts automatically? 

Second question is: because there is no account linking API, my plan is creating another account table in my database and do the linking manually, which means there are still 2 accounts in ory but only 1 in my account table.

**Comment by vinckr** [comment, marked as answer]
Hello @Yiling-J 
Ory Kratos now supports automatic account linking for OIDC.
Read more about the implementation in the official docs: https://www.ory.sh/docs/kratos/social-signin/link-multiple-provider-account and the PR https://github.com/ory/kratos/pull/3563

This is merged on master and available in production in [Ory Network](https://console.ory.sh/) and will be included in the next Ory Kratos release.
